apps/papra-server/src/modules/intake-emails/drivers/schema-username/schema-username.intake-email-driver.ts

```diff
@@ -0,0 +1,59 @@
+import type { Organization } from 'better-auth/plugins';
+import type { User } from '../../../users/users.types';
+import { createError } from '../../../shared/errors/errors';
+import { isNil } from '../../../shared/utils';
+import { getEmailUsername } from '../../intake-emails.models';
+import { defineIntakeEmailDriver } from '../intake-emails.drivers.models';
+
+export const SCHEMA_USERNAME_INTAKE_EMAIL_DRIVER_NAME = 'schema-username';
+
+export const schemaUsernameIntakeEmailDriverFactory = defineIntakeEmailDriver(
+  ({ config }) => {
+    const { domain, uniqueIdentifierSchema }
+      = config.intakeEmails.drivers.schemaUsername;
+
+    return {
+      name: SCHEMA_USERNAME_INTAKE_EMAIL_DRIVER_NAME,
+      generateEmailAddress: async (
+        userHint?: User,
+        organizationHint?: Organization,
+      ) => {
+        if (isNil(userHint) || isNil(organizationHint)) {
+          throw createError({
+            statusCode: 500,
+            message: 'User and Organization hints are required for this driver',
+            code: 'intake_emails.driver_user_organization_hints_required',
+          });
+        }
+
+        let emailIdentifier = uniqueIdentifierSchema;
+
+        // {{username}}, {{random-digits}}, {{organization-id}}, {{organization-name}}
+        const { username } = getEmailUsername({ email: userHint.email });
+        emailIdentifier = emailIdentifier.replace(
+          '{{username}}',
+          username!,
+        );
+        emailIdentifier = emailIdentifier.replace(
+          '{{random-digits}}',
+          Math.random().toString().slice(2, 5),
+        );
+        emailIdentifier = emailIdentifier.replace(
+          '{{organization-id}}',
+          organizationHint.id,
+        );
+        emailIdentifier = emailIdentifier.replace(
+          '{{organization-name}}',
+          // Organization names have quotes
+          organizationHint.name.replaceAll('"', ''),
+        );
```

Comment:
Avoid mutating variable, try to always use const instead of let
\+ using `replaceAll` in case one wanna set multiple pattern
\+ using [`slugify`](https://github.com/sindresorhus/slugify) to ensure email-safeness on the whole username

And bonus I would love the pattern generator in a standalone pure function with unit tests 

```suggestion
        const { username } = getEmailUsername({ email: userHint.email });
        
        // {{username}}, {{random-digits}}, {{organization-id}}, {{organization-name}}
        const rawEmailIdentifier = uniqueIdentifierSchema
            .replaceAll('{{username}}',  username ?? 'unknown')
            .replaceAll('{{random-digits}}', () => Math.random().toString().slice(2, 5))
            .replaceAll('{{organization-id}}', organizationHint.id)
            .replaceAll('{{organization-name}}', organizationHint.name);
        
        // https://www.npmjs.com/package/@sindresorhus/slugify
        const emailIdentifier = slugify(rawEmailIdentifier)
```